ndb_api/operation_helpers.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2022-2023 Nutanix, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ndb_api
+
+// Returns a boolean indicating if an operation has reached a terminal state
+func HasOperationFailed(o OperationResponse) bool {
+	status := o.Status
+	return status == "2" || status == "3" || status == "4"
+}
```

Comment:
Could you add comments indicating what "2", "3", and "4" mean. I understand they are failures, but might be helpful to distinguish them for future developers.